bobtimus/src/bin/bobtimus.rs

```diff
@@ -200,5 +209,27 @@ mod kraken {
 
             let _ = serde_json::from_str::<TickerUpdate>(sample_response).unwrap();
         }
+
+        #[tokio::test]
+        async fn latest_rate_does_not_wait_for_next_value() {
+            let (write, read) = watch::channel(Rate::ZERO);
+
+            let latest_rate = Rate {
+                ask: LiquidUsdt::from_str_in_dollar("20000.0").unwrap(),
+                bid: LiquidUsdt::from_str_in_dollar("19000.0").unwrap(),
+            };
+            let _ = write.broadcast(latest_rate).unwrap();
+
+            let mut service = RateService {
+                receiver: read,
+                latest_rate: Rate::ZERO,
+            };
+
+            let rate = service.latest_rate().await.unwrap();
+            assert_eq!(rate, latest_rate);
+
+            let _rate = service.latest_rate().await.unwrap();
```

Comment:
You will have to use the return value of this to make the test meaningful no?

Reply:
Haha nice catch woops